### PR TITLE
Set PROTOC and PROTOBUF_PROTOC_EXECUTABLE in CMake cache.

### PR DESCRIPTION
**Pull request by tejlmand**
cmake: update NcsConfig to set PROTOC in the CMake cache

    Jira: NCSIDB-816

    Set PROTOC and PROTOBUF_PROTOC_EXECUTABLE in CMake cache.
    Setting those variables in caches ensures that CMake code using
    find_program() for looking up 'protoc' will use the version provided
    by NCS Toolchain.

    Update nanopb to 0.4.6.
    The 0.4.6 complete release contains an updated protoc which works with
    latest protobuf python package.

    Add protobuf as required python package to ensure correct functioning
    protoc.

### CODE REVIEW

File: share/ncs-package/cmake/NcsConfig.cmake [hunk 1]
```diff
@@ -32,14 +32,23 @@ if(NOT NO_BOILERPLATE)
     find_package(NcsToolchain ${NCS_TOOLCHAIN_MINIMUM_REQUIRED} ${EXACT} QUIET)
     if(${NcsToolchain_FOUND})
       message("-- Using NCS Toolchain ${NcsToolchain_VERSION} for building. (${NcsToolchain_DIR})")
+
+      # Handle older NCS Toolchains which doesn't set NCS_TOOLCHAI_PROTOC. Those all have protoc at fixed location.
```
Comment (martingalvan-nordic) [inline share/ncs-package/cmake/NcsConfig.cmake:36]:
```suggestion
      # Handle older NCS Toolchains which doesn't set NCS_TOOLCHAIN_PROTOC. Those all have protoc at fixed location.
```

Comment (martingalvan-nordic) [reply share/ncs-package/cmake/NcsConfig.cmake:36]: Is this correct, though? Older toolchains had the issue where nanopb would be placed at different directories depending on the OS. This was `${NCS_TOOLCHAIN_PATH}/nanopb` on Windows, `${NCS_TOOLCHAIN_PATH}/opt/nanopb` on Linux, and `${NCS_TOOLCHAIN_PATH}/opt/nanopb/nanopb-0.4.4-macosx-x86`.

**Review by martingalvan-nordic — DISMISSED** [review]

**Review by vidarbe — APPROVED** [review]

File: share/ncs-package/cmake/NcsConfig.cmake (hunk 1 above)
Comment (tejlmand) [reply share/ncs-package/cmake/NcsConfig.cmake]: we are close to code freeze, so decide to clean the code and prepare for new toolchain when it's ready.

**Review by martingalvan-nordic — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -42,6 +42,11 @@ RF Front-End Modules
 
   * nRF21540 GPIO+SPI built-in power model that keeps the nRF21540's gain constant and as close to the currently configured value of gain as possible.
 
+Build system
+------------
+
+* Fixed an issue with |NCS| Toolchain where the protoc and nanopb would not be correctly detected by the build system and resulting in builds trying to find locally installed versions and not the version shipped with NCS Toolchain.
```
Comment (FrancescoSer) [inline doc/nrf/releases/release-notes-changelog.rst:48]:
```suggestion
* Fixed an issue with the |NCS| Toolchain where the protoc and nanopb would not be correctly detected by the build system, resulting in builds trying to find locally installed versions and not the version shipped with the |NCS| Toolchain.
```

**Review by FrancescoSer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (tejlmand) [reply doc/nrf/releases/release-notes-changelog.rst]: applied

**Review by FrancescoSer — APPROVED** [review]

**Pull request merged**